src/WebJobs.Script.Analyzers/DiagnosticDescriptors.cs

```diff
@@ -21,5 +21,11 @@ private static DiagnosticDescriptor Create(string id, string title, string messa
                 messageFormat: "Async void can lead to unexpected behavior. Return Task instead.",
                 category: Constants.DiagnosticsCategories.Usage,
                 severity: DiagnosticSeverity.Error);
+
+        public static DiagnosticDescriptor AvoidNonStaticHttpClient { get; }
+            = Create(id: "AZF0002", title: "Avoid non-static HttpClient usage",
+                messageFormat: "Reuse HttpClient instances to avoid holding more HTTP connections than necessary.", // TODO: fixup for code fixes?
```

Comment:
Are we recommending them to use static http client or should we be suggesting them to use `IHttpClientFactory` with DI ?